synapse/lib/node.py

```diff
@@ -668,6 +668,9 @@ async def delete(self, force=False):
             (s_layer.EDIT_NODE_DEL, (formvalu, self.form.type.stortype), ()),
         )
 
+        async for name in self.iterDataKeys():
+            edits.append((s_layer.EDIT_NODEDATA_DEL, (name, None), ()))
```

Comment:
I think we should move this above appending the EDIT_NODE_DEL, that should be the last edit performed

Reply:
Actually, it would be ideal if we could just check whether the node has a value in the write layer sode and we could skip appending these edits altogether